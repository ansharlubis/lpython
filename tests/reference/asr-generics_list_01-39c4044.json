{
    "basename": "asr-generics_list_01-39c4044",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/generics_list_01.py",
    "infile_hash": "90ee9074468153ada0f422e62b417d5dedca6f9b7ad8f7a29b2c87a5",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-generics_list_01-39c4044.stdout",
    "stdout_hash": "054f43bf709c3976f4bf4993f5d7ec05214263c68ee6666841e5ee4a",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}