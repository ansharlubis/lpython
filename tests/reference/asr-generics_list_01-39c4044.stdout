(TranslationUnit (SymbolTable 1 {T: (Variable 1 T Local () () Default (TypeParameter T [] Number) Source Public Required .false.), __lpython_generic_sum_0: (Function (SymbolTable 3 {_lpython_return_variable: (Variable 3 _lpython_return_variable ReturnVar () () Default (Integer 4 []) Source Public Required .false.), acc: (Variable 3 acc In () () Default (Integer 4 []) Source Public Required .false.), i: (Variable 3 i Local () () Default (Integer 4 []) Source Public Required .false.), lst: (Variable 3 lst In () () Default (List (Integer 4 [])) Source Public Required .false.)}) __lpython_generic_sum_0 [(Var 3 lst) (Var 3 acc)] [] [(DoLoop ((Var 3 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (ListLen (Var 3 lst) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(= (Var 3 acc) (IntegerBinOp (Var 3 acc) Add (ListItem (Var 3 lst) (Var 3 i) (Integer 4 []) ()) (Integer 4 []) ()) ())]) (= (Var 3 _lpython_return_variable) (Var 3 acc) ()) (Return)] (Var 3 _lpython_return_variable) Source Public Implementation .false. ()), _lpython_main_program: (Subroutine (SymbolTable 5 {}) _lpython_main_program [] [(Print () [(FunctionCall 1 __lpython_generic_sum_0 () [((ListConstant [(IntegerConstant 1 (Integer 4 [])) (IntegerConstant 2 (Integer 4 [])) (IntegerConstant 3 (Integer 4 [])) (IntegerConstant 4 (Integer 4 [])) (IntegerConstant 5 (Integer 4 []))] (List (Integer 4 [])))) ((IntegerConstant 0 (Integer 4 [])))] (Integer 4 []) () ())] () ())] Source Public Implementation () .false. .false.), main_program: (Program (SymbolTable 4 {}) main_program [] [(SubroutineCall 1 _lpython_main_program () [] ())]), sum: (Function (SymbolTable 2 {_lpython_return_variable: (Variable 2 _lpython_return_variable ReturnVar () () Default (TypeParameter T [] Number) Source Public Required .false.), acc: (Variable 2 acc In () () Default (TypeParameter T [] Number) Source Public Required .false.), i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.), lst: (Variable 2 lst In () () Default (List (TypeParameter T [] Number)) Source Public Required .false.)}) sum [(Var 2 lst) (Var 2 acc)] [(TypeParameter T [] Assignable)] [(DoLoop ((Var 2 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (ListLen (Var 2 lst) (Integer 4 []) ()) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) ()) (IntegerConstant 1 (Integer 4 []))) [(= (Var 2 acc) (TemplateBinOp (Var 2 acc) Add (ListItem (Var 2 lst) (Var 2 i) (TypeParameter T [] Number) ()) (TypeParameter T [] Number) ()) ())]) (= (Var 2 _lpython_return_variable) (Var 2 acc) ()) (Return)] (Var 2 _lpython_return_variable) Source Public Implementation .false. ())}) [])
